### PR TITLE
Fix back navigation to details page

### PR DESCRIPTION
**Pull request by mattiashtd**
Clicking the back button on the additional info page and on the review page often had no effect which resulted in errors afterwards.
Fix this by clicking the back button repeatedly if the navigation doesn't take place.

### CODE REVIEW

File: src/extractors.js [hunk 1]
```diff
@@ -227,13 +227,11 @@ module.exports.extractAdditionalInfo = async ({ page }) => {
         } catch (e) {
             log.info(`[PLACE]: ${e}Additional info not parsed`);
         } finally {
-            const backButton = await page.$(BACK_BUTTON_SEL);
-            if (!backButton) {
-                // eslint-disable-next-line no-unsafe-finally
-                throw new Error('Back button for additional info is not present');
+            const title = await page.$(PLACE_TITLE_SEL);
```
Comment (metalwarrior665) [inline src/extractors.js:230]: Maybe I would put this check also to the navigateBack function so it always first checks if we are on the main page (no navigation needed) and only if not then starts to wait for the back button

**Review by metalwarrior665 — CHANGES_REQUESTED** [review]
This is amazing, thanks a ton! Please just check my comment. Also, do you know how often it still completely timeouts/gets stuck?

File: src/extractors.js (hunk 1 above)
Comment (mattiashtd) [reply src/extractors.js]: Sure, fixed here: c617346dba2b447e55d0bb3837306488f51b8ced
Sorry for being late - I was on holidays :)

Comment (mattiashtd) [reply src/extractors.js]: > Also, do you know how often it still completely timeouts/gets stuck?

After the fix it didn't time out anymore for me. But I only checked it for a small amount of pages until now. Usually the click worked on the 2nd time latest. I think it never clicked 4 times or more.

Comment (mattiashtd) [reply src/extractors.js]: Oops, now really fixed with this: 6ab80a4a64491aaaf00515b4273c520b05166e48 
My brain wasn't fully back from my holidays :)

**Comment by metalwarrior665** [comment]
Thanks. Merging it in. We will test it a bit in beta on Apify.

**Pull request merged**